go/adbc/sqldriver/driver_test.go

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build cgo
+
+package sqldriver_test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/apache/arrow-adbc/go/adbc/drivermgr"
+	"github.com/apache/arrow-adbc/go/adbc/sqldriver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSqlDriver(t *testing.T) {
+	sql.Register("adbc", sqldriver.Driver{&drivermgr.Driver{}})
+
+	db, err := sql.Open("adbc", "driver=adbc_driver_sqlite;entrypoint=AdbcDriverInit")
+	require.NoError(t, err)
+	assert.NotNil(t, db)
+
+	rows, err := db.Query("SELECT 1")
+	assert.NoError(t, err)
+	defer rows.Close()
+
+	fmt.Println(rows.Columns())
+	cols, err := rows.ColumnTypes()
+	require.NoError(t, err)
+	fmt.Println(cols[0].Name(), cols[0].DatabaseTypeName())
+	fmt.Println(cols[0].Nullable())
+
+	for rows.Next() {
+		var v int64
+		if err := rows.Scan(&v); err != nil {
+			t.Fatal(err)
+		}
+		fmt.Println(v)
```

Comment:
Should this actually assert on the value? And should we assert we actually enter the loop?

Reply:
yea, i should be asserting on this instead of just printing it, i'll fix that.

Reply:
Actually, i have a better Idea. I'll make this a testable example that would show up in the docs and verify it's output. That way we get a solid example in the pkg.go.dev docs for this